From d4e4137dccd004a8a07cae95903afcd90c588768 Mon Sep 17 00:00:00 2001
From: wbs306 <wbs306@gmail.com>
Date: Thu, 27 Oct 2022 20:58:17 +0800
Subject: [PATCH 8/8] Camera: Expose aux camera if packagename is null

In case of the pacckage name not being found properly
the name would always be found in the excludelist, even
if the list is empty. This leads to denying aux camera
access in some cases where it is not intended to be
blocked.

Change-Id: I75c620ab876ebabf71c740e04004e4120edd93ae
---
 core/java/android/hardware/Camera.java | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/java/android/hardware/Camera.java b/core/java/android/hardware/Camera.java
index 2fbcf5bf..7e59d860 100644
--- a/core/java/android/hardware/Camera.java
+++ b/core/java/android/hardware/Camera.java
@@ -278,6 +278,8 @@ public class Camera {
          * if the package name does not falls in this bucket
          */
         String packageName = ActivityThread.currentOpPackageName();
+        if (packageName == null)
+            return true;
         List<String> packageList = new ArrayList<>(Arrays.asList(
                 SystemProperties.get("vendor.camera.aux.packagelist", ",").split(",")));
         List<String> packageExcludelist = new ArrayList<>(Arrays.asList(
-- 
2.40.0

